timesketch/lib/analyzers/sigma_tagger.py

```diff
@@ -71,64 +46,41 @@ def run(self):
-                    rule_file_path = os.path.join(dirpath, rule_filename)
-                    rule_file_path = os.path.abspath(rule_file_path)
-                    logger.info('[sigma] Reading rules from {0!s}'.format(
-                        rule_file_path))
-                    with codecs.open(rule_file_path, 'r', encoding='utf-8',
-                                     errors='replace') as rule_file:
-                        try:
-                            rule_file_content = rule_file.read()
-                            parser = sigma_collection.SigmaCollectionParser(
-                                rule_file_content, self.sigma_config, None)
-                            parsed_sigma_rules = parser.generate(sigma_backend)
-                        except NotImplementedError as exception:
-                            logger.error(
-                                'Error generating rule in file {0:s}: {1!s}'
-                                .format(rule_file_path, exception))
-                            continue
-
-                        for sigma_rule in parsed_sigma_rules:
-                            try:
-                                sigma_rule_counter += 1
-                                logger.info(
-                                    '[sigma] Generated query {0:s}'
-                                    .format(sigma_rule))
-                                tagged_events_counter = self.run_sigma_rule(
-                                    sigma_rule, tag_name)
-                                tags_applied[tag_name] += tagged_events_counter
-                            except elasticsearch.TransportError as e:
-                                logger.error(
-                                    'Timeout generating rule in file {0:s}: '
-                                    '{1!s} waiting for 10 seconds'.format(
-                                        rule_file_path, e), exc_info=True)
-                                time.sleep(10) # waiting 10 seconds
+        sigma_rules = ts_sigma_lib.get_all_sigma_rules()
+
+        if sigma_rules is None:
+            logger.error('No  Sigma rules found. Check SIGMA_RULES_FOLDERS')
+
+        problem_string = []
+
+        for rule in sigma_rules:
+            tags_applied[rule.get('file_name')] = 0
+            try:
+                sigma_rule_counter += 1
+                tagged_events_counter = self.run_sigma_rule(
+                    rule.get('es_query'), rule.get('file_name'))
+                tags_applied[rule.get('file_name')] += tagged_events_counter
+            except elasticsearch.TransportError as e:
+                logger.error(
+                    'Timeout executing search for {0:s}: '
+                    '{1!s} waiting for 10 seconds'.format(
+                        rule.get('file_name'), e), exc_info=True)
+                # this is caused by to many ES queries in short time range
+                # thus waiting for 10 seconds before sending the next one.
```

Comment:
the comment says `wait for 10 seconds` but the code says 1 sec

Reply:
ah tested that, back to 10...